src/java/org/apache/cassandra/db/marshal/DynamicCompositeType.java

```diff
@@ -100,6 +102,14 @@ private DynamicCompositeType(Map<Byte, AbstractType<?>> aliases)
             this.inverseMapping.put(en.getValue(), en.getKey());
     }
 
+    @Override
+    public DynamicCompositeType overrideKeyspace(Function<String, String> overrideKeyspace)
+    {
+        return new DynamicCompositeType(aliases.entrySet().stream()
```

Comment:
see https://github.com/datastax/cassandra/pull/331/files#r801303769, we should use getInstance to reduce GC.